common/src/main/java/com/strategyobject/substrateclient/common/inject/Inject.java

```diff
@@ -0,0 +1,6 @@
+package com.strategyobject.substrateclient.common.inject;
+
+public interface Inject<R, D> {
```

Comment:
I guess `R` should extend the interface.

Reply:
That would imply that the injection produces something that you can inject into again, which is not true. The opposite, I want to encourage to call inject only once. Currently, you can inject multiple `optionWriter.inject(...).inject(...).inject(...)` and according to the current implementation only the first injection is used, all the rest are ignored. So, I think it is not right to put it into the contract that R extends Inject.